mysql/infoschema.go

```diff
@@ -301,9 +303,126 @@ func getConstraints(conv *internal.Conv, db *sql.DB, table schemaAndName) ([]str
 			m[col] = append(m[col], constraint)
 		}
 	}
-	return primaryKeys, m, nil
+	// Get all the foreign keys for the table
+	refTables, _ := getRefTables(db, table)
+	for _, refTable := range refTables {
+		fkey, _ := getForeignKey(conv, db, table, refTable)
+		ForeignKeys = append(ForeignKeys, fkey)
+	}
+	return primaryKeys, m, ForeignKeys, nil
+}
+
+// getRefTables return the list of referenced tables for
+// the selected database and table.
+func getRefTables(db *sql.DB, table schemaAndName) ([]schemaAndName, error) {
+	q := "select REFERENCED_TABLE_NAME FROM INFORMATION_SCHEMA.REFERENTIAL_CONSTRAINTS WHERE CONSTRAINT_SCHEMA=? and TABLE_NAME=?;"
+	rows, err := db.Query(q, table.schema, table.name)
+	if err != nil {
+		return nil, fmt.Errorf("couldn't get reference tables: %w", err)
+	}
+	defer rows.Close()
+	var refTableName string
+	var refTables []schemaAndName
+	for rows.Next() {
+		rows.Scan(&refTableName)
+		refTables = append(refTables, schemaAndName{schema: table.schema, name: refTableName})
+	}
+	return refTables, nil
+}
+
+// getForeignKey returns the foreign key constraint for
+// a particular referenced table in the selected table
+// and database.
+func getForeignKey(conv *internal.Conv, db *sql.DB, table schemaAndName, refTable schemaAndName) (schema.Fkey, error) {
```

Comment:
Thanks for laying out the two options.
Definitely agree with you that the first option is better! Please go ahead with that approach.